apps/backend/src/factory/StfcDownloadService.ts

```diff
@@ -0,0 +1,149 @@
+import querystring from 'querystring';
+import { Readable } from 'stream';
+import { ReadableStream } from 'stream/web';
+
+import { logger } from '@user-office-software/duo-logger';
+import contentDisposition from 'content-disposition';
+import { Request, Response, NextFunction } from 'express';
+import { container } from 'tsyringe';
+
+import { Tokens } from '../config/Tokens';
+import { FileDataSource } from '../datasources/IFileDataSource';
+import { Role } from '../models/Role';
+import {
+  MetaBase,
+  DownloadType,
+  PDFType,
+  XLSXType,
+  ZIPType,
+  DownloadService,
+} from './DownloadService';
+import { ProposalPDFData } from './pdf/proposal';
+
+const ENDPOINT = process.env.USER_OFFICE_FACTORY_ENDPOINT;
+
+export class StfcDownloadService implements DownloadService {
+  async callFactoryService<TData, TMeta extends MetaBase>(
+    downloadType: DownloadType,
+    type: PDFType | XLSXType | ZIPType,
+    properties: { data: TData[]; meta: TMeta; userRole: Role },
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    if (!ENDPOINT) {
+      logger.logError(
+        'Could not start application: the `USER_OFFICE_FACTORY_ENDPOINT` environment variable is missing. Exiting.',
+        {}
+      );
+      process.exit(1);
+    }
+
+    let isPdfAvailable = false;
+    let proposalPdfData = null;
+    const data = properties?.data[0] as ProposalPDFData;
+
+    if (properties?.data?.length === 1) {
+      const fileDataSource = container.resolve<FileDataSource>(
+        Tokens.FileDataSource
+      );
+      proposalPdfData = await fileDataSource.getBlobdata(
+        `Migrated-${data.proposal.proposalId}.pdf`
```

Comment:
Sorry to only realise this now, but I think unfortunately there needs to be a way of fetching the reference number for a specific facility. Because there are some reference numbers that are shared between facilities. For example, 91012 is a reference number for both LSF and HPL.

Reply:
I have done the changes to save the pdf with facility name included in file name. Please have a look.